backend/datasource/serve_example_test.go

```diff
@@ -61,15 +61,23 @@ func (ds *testDataSource) CheckHealth(ctx context.Context, req *backend.CheckHea
 	}
 
 	// Handle request
-	_, _ = settings.httpClient.Get("http://")
+	resp, err := settings.httpClient.Get("http://")
+	if err != nil {
+		return nil, err
+	}
+	resp.Body.Close()
 	return nil, nil
 }
 
 func (ds *testDataSource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
 	var resp *backend.QueryDataResponse
 	err := ds.im.Do(req.PluginContext, func(settings *testDataSourceInstanceSettings) error {
 		// Handle request
-		_, _ = settings.httpClient.Get("http://")
+		resp, err := settings.httpClient.Get("http://")
```

Comment:
I love that linters catch this potential bug. So easy to make, yet important to avoid. But I'm not sure it makes sense for tests? In an example, it makes sense since it shows the consumers that its important. 

NB: In the early days of Prometheus at Soundcloud they took down all there infra monitored by Prometheus by not closing the body and therefore consuming all connections on the servers being monitored :) It's in some promcon talk from 2016.

Reply:
@bergquist Yeah I also thought it might not matter for tests, the question is how to disable it for those cases. It was easier to leave it on, and I think in examples it makes perfect sense :)

Reply:
Another bug the stricter linting actually caught is using a loop variable in `t.Run` closures. If I'm not totally wrong here (and the linter suggests I'm right), the `t.Run` closures would potentially all use the same value of the loop variable (depending on when they execute).